### PR TITLE
Remove gl resize

### LINKED ISSUE #3114: Refactor gl.canvas to use clientWidth/Height instead of relying on external calls to resizeCanvas
From WebGL anti patterns entry 4: https://webglfundamentals.org/webgl/lessons/webgl-anti-patterns.html

Example code:
```
function resizeCanvasToDisplaySize() {
  var width = gl.canvas.clientWidth;
  var height = gl.canvas.clientHeight;
  if (gl.canvas.width != width ||
      gl.canvas.height != height) {
     gl.canvas.width = width;
     gl.canvas.height = height;
  }
}

function render() {
   resizeCanvasToDisplaySize();
   drawStuff();
   requestAnimationFrame(render);
}
render();
```

This will also remove some of the reliance on DOM calls in the engine that we shouldn't really be using.

Also, this should help fix some of the rendering stretching bugs we have on the Samsung browser and iOS browser.

### PR DESCRIPTION
**Pull request by yaustar**
Fixes #3114

Tested against model-outline, WebXR VR and AR projects as well as other private projects that previously ran into issues on the release of 1.41.0

Edit Sun 9 May 2021: Tested against water reflection demo: https://launch.playcanvas.com/1144354?debug=true ([forum thread](https://forum.playcanvas.com/t/solved-how-to-determine-location-entity-name-that-used-deleted-texture/19938/))

I confirm I have read the [contributing guidelines](https://github.com/playcanvas/engine/blob/master/.github/CONTRIBUTING.md) and signed the [Contributor License Agreement](https://docs.google.com/a/playcanvas.com/forms/d/1Ih69zQfJG-QDLIEpHr6CsaAs6fPORNOVnMv5nuo0cjk/viewform).

### CODE REVIEW

**Comment by Maksims** [comment]
Not a big fan of `updateGlCanvasSize` naming. `updateCanvasSize` would do as good?

**Comment by yaustar** [comment]
Actually, I take that back. `updateCanvasSize` does make sense after sitting on it for a bit.

**Review by mvaligursky — APPROVED** [review]

**Comment by yaustar** [comment]
Planning a little ahead here. This function will need to be called when the resize and orientation events are fired for some older projects to works as they depend on the canvas resolution being correct after a resize event.

File: src/graphics/graphics-device.js [hunk 1]
```diff
@@ -3496,8 +3497,8 @@ class GraphicsDevice extends EventHandler {
         this._height = height;
 
         var ratio = Math.min(this._maxPixelRatio, window.devicePixelRatio);
-        width *= ratio;
-        height *= ratio;
+        width = Math.floor(width * ratio);
+        height = Math.floor(height * ratio);
 
         if (this.canvas.width === width && this.canvas.height === height)
             return;
```
Comment (mvaligursky) [inline src/graphics/graphics-device.js:3504]: while you're here, maybe flip the logic and get rid of the return.

**Review by mvaligursky — APPROVED** [review]

File: src/graphics/graphics-device.js (hunk 1 above)
Comment (yaustar) [reply src/graphics/graphics-device.js]: Done!

Comment (mvaligursky) [reply src/graphics/graphics-device.js]: perfect.

**Pull request merged**